src/BuildingRegistry.Consumer.Address/ConsumerAddressContext.cs

```diff
@@ -18,6 +22,42 @@ public ConsumerAddressContext(DbContextOptions<ConsumerAddressContext> options)
         { }
 
         public override string ProjectionStateSchema => Schema.ConsumerAddress;
+
+        public AddressData? GetOptional(AddressPersistentLocalId addressPersistentLocalId)
+        {
+            var item = AddressConsumerItems
+                .AsNoTracking()
+                .SingleOrDefault(x => x.AddressPersistentLocalId == addressPersistentLocalId);
```

Comment:
Don't we want to use Find() instead of SingleOrDefault?

Reply:
You can combine them in another way, but I think `SingleOrDefault` is not bad considering `Find` will very rarely find one in localcache.